### ISSUE TITLE
Exception : Relation jv_commit already exists in JaVers 5.8.8

### ISSUE THREAD

**Issue by nikjain10** [javers/javers#934]
**I am using JaVers version 5.8.8, Spring boot 2.1.5, PostgreSQL 42.2.8. Its running fine and 4 tables got created as expected when application runs first time. However error comes when application starts second time and so on. It is mentioned in official documentation of JaVers that this issue was fixed in version 5.6.2 but i am using latest version 5.8.8 still this error persists. Below is the error stacktrace: -**

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
[2m2020-01-21 18:30:04.506[0;39m [31mERROR[0;39m [35m59224[0;39m [2m---[0;39m [2m[  restartedMain][0;39m [36mo.s.boot.SpringApplication              [0;39m [2m:[0;39m Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'auditController' defined in file [D:\billing-system-principal-service\target\classes\com\billing\controller\AuditController.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'JaversFromStarter' defined in class path resource [org/javers/spring/boot/sql/JaversSqlAutoConfiguration.class]: Invocation of init method failed; nested exception is org.polyjdbc.core.exception.SchemaManagerException: [DDL_ERROR] Failed to run DDL:
CREATE TABLE jv_commit (
commit_pk BIGINT NOT NULL,
author VARCHAR(200),
commit_date TIMESTAMP,
commit_date_instant VARCHAR(30),
commit_id NUMERIC(22,2),
CONSTRAINT jv_commit_pk PRIMARY KEY(commit_pk)
) 
    at

**Comment by bartoszwalacik** [comment]
please push a failing test case with the reproduction of this issue

**Comment by nikjain10** [comment]
@bartoszwalacik  I am not sure about the test case. Do you require steps of creating and executing application to reproduce this ?

**Comment by bartoszwalacik** [comment]
yes, see https://github.com/javers/javers/tree/master/javers-persistence-sql/src/test/groovy/org/javers/repository/sql/integration/docker

**Comment by nikjain10** [comment]
Following are the steps which i performed in my spring boot application : 

1) Included following in pom.xml : -

                **<dependency>
			<groupId>org.javers</groupId>
			<artifactId>javers-spring-boot-starter-sql</artifactId>
			<version>5.8.8</version>
		</dependency>**

2) Include annotation @JaversSpringDataAuditable in my repository class.

Then i started my application, 4 tables got created and successfully application got started. Later i stopped my application and then again started but this time i started getting this exception and application didnt start successfully.

**Comment by bartoszwalacik** [comment]
as i said, You are supposed to write a runnable test case which would reproduce your issue

**Comment by nikjain10** [comment]
I dont know how to write test case for this. Issue is occurring after server is restarted.  Currently i have configured **javers.sqlSchemaManagementEnabled=false** in application.properties file to make the server started.

**Comment by bartoszwalacik** [comment]
PostgreSQL 42.2.8? how is that possible if the latest version of PostrgeSQL is 12.1

**Comment by nikjain10** [comment]
yeah, to be more specific I am using below dependency in Spring boot and database version is 11.4

                 **<dependency>
			<groupId>org.postgresql</groupId>
			<artifactId>postgresql</artifactId>
			<version>42.2.8</version>
		</dependency>**

**Comment by bartoszwalacik** [comment]
@nikjain10 here is the dockerized test running on the latest postgres server (12.1).
This test restarts javers 103 times on the same database. It works. Run it on your local and check.

If you can't prove a bug by showing code - i can't help you.

https://github.com/javers/javers/blob/master/javers-persistence-sql/src/test/groovy/org/javers/repository/sql/integration/docker/PostgresDockerIntegrationTest.groovy

**Comment by nikjain10** [comment]
Could you please also check with postgre server 11.4 ?

**Comment by bartoszwalacik** [comment]
you can do it

**Comment by bartoszwalacik** [comment]
please reopen when you come up with a runnable test case

**Comment by martinplank** [comment]
I have the same problem, when using other than public schema in PostgreSQL.
@bartoszwalacik Is it possible to specify schema name for `TableNameProvider`, when using Spring Boot integration?

**Comment by KaisNeffati** [comment]
Same issue for javers 5.8.10

**Comment by bartoszwalacik** [comment]
I will reopen this issue when you come up with a runnable test case

https://github.com/javers/javers/issues/934#issuecomment-577646922

**Comment by jarekczek** [comment]
Seems like I have the same problem on Oracle 12 with custom schema name. How can I provide test case for Oracle?

**Comment by bartoszwalacik** [comment]
@jarekczek see https://github.com/javers/javers/blob/master/javers-persistence-sql/src/test/groovy/org/javers/repository/sql/integration/docker/OracleDockerIntegrationWithSchemaTest.groovy

**Comment by walkingcoding** [comment]
With version 6.6.3, I have the same problem on Mysql 8 with custom schema name. Frist start application success init table to schema. when restart throw exceptions Table xxx already exists. 
How can I do it?
This is my configuration:
-----------
javers:
  sqlSchema: data_audit
  sqlGlobalIdTableName: w_global_id
  sqlCommitTableName: w_commit
  sqlSnapshotTableName: w_snapshot
  sqlCommitPropertyTableName: w_commit_property

**Comment by bartoszwalacik** [comment]
Could you provide a failing test case? See tests in https://github.com/javers/javers/tree/master/javers-spring-boot-starter-mongo/src/test/groovy/org/javers/spring/boot/mongo

**Comment by Nosyass** [comment]
same problem

Mapping the javer tables and restarting the application causes this error.

what type of test you want to have you can reproduce the case simply with a blank application and the spring boot javers dependency
Caused by: org.postgresql.util.PSQLException: ERROR: relation 'jv_global_id' already exists

Even with this DDL configuration
   jpa:
     show-sql: true
     hibernate:
       ddl-auto: validate

**Comment by bartoszwalacik** [comment]
@Nosyass we don't use code snippets nor images thrown into comments, please try to proof your stance by providing a failing test case. Note there is a test 
https://github.com/javers/javers/blob/b263ea883c148b50c4f6dfb591b2241a88ea591c/javers-persistence-sql/src/test/groovy/org/javers/repository/sql/JaversSqlRepositoryE2ETest.groovy#L101
running on each db, which shows that the restart of javers works fine. Also you can use this test to debug the problem on your DB

**Comment by lwuethrich-devedis** [comment]
@bartoszwalacik The ddl error seems to be fixed by @Nosyass answer. If using a specific schema (for me on postgresql) creates the necessary tables correctly in the given schema but it does not seem to recognize that the tables exist in the schema when starting up

**Comment by bartoszwalacik** [comment]
Thanks for the info @lwuethrich-devedis , could you please come up with a failing test case ?(see my previous comment). It would greatly help us to spot the bug

**Comment by lwuethrich-devedis** [comment]
@bartoszwalacik I am not using Javers currently and do not have the resources to do so. My comment was to support the previously given solution to the problem and to indicate that someone else has the same issue.
